besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1502,10 +1505,46 @@ private BesuCommand configure() throws Exception {
         .ifPresent(p -> ensureAllNodesAreInAllowlist(staticNodes, p));
     metricsConfiguration = metricsConfiguration();
 
+    if (isGoQuorumCompatibilityMode) {
+      configureGoQuorumPrivacy();
+    }
+
     logger.info("Security Module: {}", securityModuleName);
     return this;
   }
 
+  private void configureGoQuorumPrivacy() {
+
+    GoQuorumPrivacyParameters.isEnabled = true;
+    final EnclaveFactory enclaveFactory = new EnclaveFactory(Vertx.vertx());
+    if (privacyKeyStoreFile != null) {
+      GoQuorumPrivacyParameters.goQuorumEnclave =
+          enclaveFactory.createGoQuorumEnclave(
+              privacyUrl,
+              privacyKeyStoreFile,
+              privacyKeyStorePasswordFile,
+              privacyTlsKnownEnclaveFile);
+    } else {
+      GoQuorumPrivacyParameters.goQuorumEnclave = enclaveFactory.createGoQuorumEnclave(privacyUrl);
+    }
+    final String key;
+    try {
+      key = Files.asCharSource(privacyPublicKeyFile, UTF_8).read();
+    } catch (final Exception e) {
+      throw new ParameterException(
+          this.commandLine,
+          "--privacy-public-key-file must be set when goquorum-compatibility-enabled is set to true.",
+          e);
+    }
+    if (key.length() != 44) {
+      throw new IllegalArgumentException(
+          "Contents of enclave public key file needs to be 44 characters long to decode to a valid 32 byte public key.");
+    }
+    // throws exception if invalid base 64
+    Base64.getDecoder().decode(key);
```

Comment:
NIT: It would be nice to encapsulate all this logic to read the key into its own method. Something like `readEnclaveKey` or similar.

Reply:
Done